src/wp-includes/rest-api/endpoints/class-wp-rest-font-library-controller.php

```diff
@@ -0,0 +1,453 @@
+			$font   = new WP_Font_Family( $font_data );
+			$result = $font->uninstall();
+			if ( is_wp_error( $result ) ) {
+				$errors[] = $result;
+			} else {
+				$successes[] = $result;
+			}
+		}
+		$data = array(
+			'successes' => $successes,
+			'errors'    => $errors,
+		);
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Checks whether the user has permissions to update the Font Library.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return true|WP_Error True if the request has write access for the item, WP_Error object otherwise.
+	 */
+	public function update_font_library_permissions_check() {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_cannot_update_font_library',
+				__( 'Sorry, you are not allowed to update the Font Library on this site.' ),
+				array(
+					'status' => rest_authorization_required_code(),
+				)
+			);
+		}
+		return true;
+	}
+
+	/**
+	 * Checks whether the user has write permissions to the temp and fonts directories.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return true|WP_Error True if the user has write permissions, WP_Error object otherwise.
+	 */
+	private function has_write_permission() {
+		// The update endpoints requires write access to the temp and the fonts directories.
+		$temp_dir   = get_temp_dir();
+		$upload_dir = WP_Font_Library::get_fonts_dir();
+		if ( ! is_writable( $temp_dir ) || ! wp_is_writable( $upload_dir ) ) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Checks whether the request needs write permissions.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array[] $font_families Font families to install.
+	 * @return bool Whether the request needs write permissions.
+	 */
+	private function needs_write_permission( $font_families ) {
+		foreach ( $font_families as $font ) {
+			if ( isset( $font['fontFace'] ) ) {
+				foreach ( $font['fontFace'] as $face ) {
+					// If the font is being downloaded from a URL or uploaded, it needs write permissions.
+					if ( isset( $face['downloadFromUrl'] ) || isset( $face['uploadedFile'] ) ) {
+						return true;
+					}
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Installs new fonts.
+	 *
+	 * Takes a request containing new fonts to install, downloads their assets, and adds them
+	 * to the Font Library.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param WP_REST_Request $request The request object containing the new fonts to install
+	 *                                 in the request parameters.
+	 * @return WP_REST_Response|WP_Error The updated Font Library post content.
+	 */
+	public function install_fonts( $request ) {
+		// Get new fonts to install.
+		$fonts_param = $request->get_param( 'font_families' );
+
+		/*
+		 * As this is receiving form data, the font families are encoded as a string.
+		 * The form data is used  because local fonts need to use that format to
+		 * attach the files in the request.
+		 */
+		$fonts_to_install = json_decode( $fonts_param, true );
```

Comment:
I don't understand this, why are we passing json as a string parameter instead of the actual data we want?

Reply:
Because we are sending FormData and not JSON data, FormData format is required to upload font files from the users' devices.

Reply:
The REST API accepts form data as input natively. The REST API controllers shouldn't wrap their own JSON in inputs instead.